RentACar/RentACar/Repositories/ClientRepository.cs

```diff
@@ -10,18 +11,38 @@ public class ClientRepository : Repository<Client>, IClientRepository
     {
         public ClientRepository(FilePathsStorage storageConfig) : base(storageConfig.Client) { }
 
-        public override Client Create(Client entity)
+        public override Client Create(Client newEntity)
         {
-            if (EntityList.Any(e => e.Dni == entity.Dni))
+            if (DniExistInList(newEntity.Dni))
                 return null;
 
-            entity.LastUpdate = DateTime.UtcNow;
+            newEntity.LastUpdate = DateTime.UtcNow;
 
-            var clientCreated = base.Create(entity);
+            var clientCreated = base.Create(newEntity);
 
             return clientCreated;
         }
 
+        public override Client Update(Client newEntity, int id)
+        {
+            var existingEntity = Get(id);
+
+            if (existingEntity is null)
+                return null;
+
+            if ( existingEntity.Dni != newEntity.Dni)
+            {
+                if (DniExistInList(newEntity.Dni))
+                    return null;
+            }
+
+            UpdateEntity(existingEntity, newEntity);
+
+            CrudHelper<Client>.SaveListToFile(EntityList, _jsonFile);
+
+            return existingEntity;
+        }
```

Comment:
Honesty, Im not too happy with this implementation, I know I'm not reusing the code, but I need to check if Dni exists in other clients before persisting, but after checking if id is null. So i changed the access modifier of _jsonFile and i override 100% of method. Another option maybe after checking if Dni already exists, call the base method, but in that case it would get and check the Client twice.
I think in the future when we have the service layer this validation will be there.. or not? 🙄

Reply:
yes, I think this could also fit in a service layer but I would just made the same you did here (plus adding a unique constraint in the database)

Reply:
nice! good to know! thanks!